### ISSUE TITLE
[Bug] FATA[0000] Parse config error: proxy 0: key '' missing  

### ISSUE THREAD

**Issue by Molay** [Dreamacro/clash#1718]
### Verify steps

- [ ]  如果你可以自己 debug 并解决的话，提交 PR 吧 Is this something you can **debug and fix**? Send a pull request! Bug fixes and documentation fixes are welcome. 
- [X]  我已经在 [Issue Tracker](……/) 中找过我要提出的问题 I have searched on the [issue tracker](……/) for a related issue. 
- [ ]  我已经使用 dev 分支版本测试过，问题依旧存在 I have tested using the dev branch, and the issue still exists. 
- [X]  我已经仔细看过 [Documentation](https://github.com/Dreamacro/clash/wiki/) 并无法自行解决问题 I have read the [documentation](https://github.com/Dreamacro/clash/wiki/) and was unable to solve the issue. 
- [X]  这是 Clash 核心的问题，并非我所使用的 Clash 衍生版本（如 OpenClash、KoolClash 等）的特定问题 This is an issue of the Clash core *per se*, not to the derivatives of Clash, like OpenClash or KoolClash. 

### Clash version

higher than 2021.09.15

### What OS are you seeing the problem on?

Linux

### Clash config

```yaml
with general config (from https://github.com/Dreamacro/clash/wiki/configuration)
```

### Clash log

```shell
FATA[0000] Parse config error: proxy 0: key '' missing
```

### Description

A fatal exception occurred during 'docker run' startup with the current latest aarch64 version (>2021.09.15) of 'dreamacro/clash-premium' image.
The version '2021.09.15' and previous versions are normal.
This problem seems to be the parsing problem of the configuration file.

**Comment by fakeboboliu** [comment]
Fixed in #1705 / #1719

**Comment by Molay** [comment]
Thanks
